src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.tt

```diff
@@ -0,0 +1,104 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
```

Comment:
Help me understand the purpose of using this technology?

Reply:
It was actually super useful yesterday when I changed the signature of all the verbed method with one line of change in the .tt (the serviceName shoud have been a string? from the beginning)

Reply:
I haven't seen this in any of our other projects, now when someone has to work in this project, that person will have to understand this technology.
If this is so helpful, we should agree across the team to use it, not in a one-off manner.

Reply:
It's amazingly helpful, but you is meta => requires to think in terms of models and code generation. It's helpful when you have a do a lot of things of the same kind. it avoids copy/paste, and also ensures consistency. It help to evolve things consistency and efficiently.
Now ... it might not be for everybody.
In the end it generates code. You don't need to touch the .tt once the code is generated, until you want to evolve things. Note the usage of partial classes here.